### PR TITLE
fixing returned error code for network creation from 500 to 409

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <yasstij11@gmail.com>

**- What I did** fixed status code from 500 to 409 see #34459

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by ripcurld0** [comment]
This closes #34459

File: api/server/router/network/network_routes.go [hunk 1]
```diff
@@ -219,7 +219,7 @@ func (n *networkRouter) postNetworkCreate(ctx context.Context, w http.ResponseWr
 			// check if user defined CheckDuplicate, if set true, return err
 			// otherwise prepare a warning message
 			if create.CheckDuplicate {
-				return libnetwork.NetworkNameError(create.Name)
+				return httputils.WriteJSON(w, http.StatusConflict, libnetwork.NetworkNameError(create.Name))
```
Comment (ripcurld0) [inline api/server/router/network/network_routes.go]: I think the last argument in `httputils.WriteJSON` should be `libnetwork.NetworkNameError(create.Name).Error()`.
However I might be wrong.

Comment (yastij) [inline api/server/router/network/network_routes.go]: Typo

**Review by boaz0 — CHANGES_REQUESTED** [review]
I think it will be lovely if there are integration tests to this change.

File: api/server/router/network/network_routes.go (hunk 1 above)
Comment (yastij) [inline api/server/router/network/network_routes.go]: Fixed

**Comment by yastij** [comment]
@ripcurld0 - integration tests should be added in `server_test.go` ?

**Comment by ripcurld0** [comment]
More likely to [integration-cli/docker_api_network_test.go](https://github.com/moby/moby/blob/master/integration-cli/docker_api_network_test.go#L46-L71).

**Comment by yastij** [comment]
thanks for the hint.

**Comment by yastij** [comment]
@ripcurld0 - we should just [https://github.com/moby/moby/blob/master/integration-cli/docker_api_network_test.go#L315](this) to check precisely against conflict code I think.

**Comment by ripcurld0** [comment]
@yastij if some tests fail because of this, then we will have to change it but my guess is it's fine and no need to be changed. 🥇

**Comment by yastij** [comment]
@ripcurld0 this assertion already checks against duplicate network, do you prefer to have another that checks specifically for the conflict status ?

**Comment by tossmilestone** [comment]
I think `api/server/router/network/network_routes.go:212` should also be changed:
```golang
 	if nws, err := n.cluster.GetNetworksByName(create.Name); err == nil && len(nws) > 0 {
 		return libnetwork.NetworkNameError(create.Name)
 	}
```
WDYT@yastij @ripcurld0

**Comment by ripcurld0** [comment]
@tossmilestone yes that's right. @yastij  we have to update that line too.

**Comment by yastij** [comment]
Sure, taking care of it by tomorrow !

**Comment by cpuguy83** [comment]
Please return an error using the new errdefs packag: https://godoc.org/github.com/moby/moby/api/errdefs

**Comment by thaJeztah** [comment]
This one is carried by @tossmilestone  in https://github.com/moby/moby/pull/35030

**Pull request closed without merging**